### DISCUSSION TITLE
Re-mapping controls

### DISCUSSION THREAD

**Discussion by curif** [curif/AgeOfJoy-2022.1#42] [Ideas]
Not all games needs the same controls. We will need a way to customize controls for each game.

The idea is to improve `CDL` to write the re-mapping, adding configurations to change the normal behavior of the Quest controls. We can configure the re-mapping for each cabinet, and if a user don't like it, he can change it easily. Or copy the cabinet, change its name and use the same cabinet but with different control behavior.

**Comment by WilsonBradley** [comment]
**Please make Re-mapping Controls high on request list..
I always accidentally press Left joy controller button (push in) causing the MAME menu to pop up while playing.
Most importantly, I am right handed. Right joy would be ideal.
Always forget how to detach from Cabinet to move around again.
All these, make it not really playable yet..

I think it's time to start adding a Start Menu when starting the actual AgeOfJoy - (more details in other post)
Example;
                                          **Menu**
                                              Start 
                                           Settings
                               Control assignments
                             Map of rooms / games 
                                                Help 
(- include Controller layout if still not configurable yet. 
https://img.itch.zone/aW1nLzEwNjM3Mjc1LnBuZw==/original/2kNczL.png

**Comment by RaulRomanillos** [comment]
Hello, my name is Raul Romanillos, and I'd like to know the steps for mapping the keystrokes on an iCade arcade controller to keyboard strokes for use in a game, such as Ghost's Goblins or R-Type.
Thanks and best regards.
